evap/evaluation/tests/test_views.py

```diff
@@ -246,3 +248,45 @@ def test_notebook(self):
 
         user.refresh_from_db()
         self.assertEqual(user.notes, self.note)
+
+
+class TestResetEvaluation(WebTestStaffMode):
+
+    @classmethod
+    def setUpTestData(cls) -> None:
+        cls.manager = make_manager()
+        cls.semester = baker.make(Semester, results_are_archived=True)
+
+    def reset_from_x_to_new(self, x, success_expected: bool) -> None:
+        evaluation = baker.make(Evaluation, state=x, course__semester=self.semester)
+
+        semester_overview_page = self.app.get(
+            reverse("staff:semester_view", args=[self.semester.pk]), user=self.manager, status=200
+        )
+
+        form = semester_overview_page.forms["evaluation_operation_form"]
+
+        form["evaluation"] = [evaluation.pk]
+
+        confirmation_page = form.submit(name="target_state", value=str(Evaluation.State.NEW))
+
+        if success_expected:
+            confirmation_form = confirmation_page.forms["evaluation-operation-form"]
+            confirmation_form.submit()
+            self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).state, Evaluation.State.NEW)
+        else:  # no confirmation form should exist
+            self.assertNotIn("evaluation-operation-form", confirmation_page.forms)
```

Comment:
(Nit) I think we can be more succinct here:
```suggestion
        if success_expected:
            confirmation_page.forms["evaluation-operation-form"].submit()
            self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).state, Evaluation.State.NEW)
        else:
            self.assertNotIn("evaluation-operation-form", confirmation_page.forms)
```